src/Adapter/Product/Update/Filler/BasicInformationFiller.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Update\Filler;
+
+use PrestaShop\PrestaShop\Adapter\Tools;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\UpdateProductCommand;
+use Product;
+
+/**
+ * Fills product properties which can be considered as a basic product information
+ */
+class BasicInformationFiller implements ProductFillerInterface
```

Comment:
I don't understand the whole point of having `*FillerInterface` files and classes.
Why don't we fetch the product and map data inside ?

Reply:
purely just to keep them small I guess. Cuz we will have like 5-6 of those classes (and we already had all the logic taken from dedicated command handlers that used it before). 
Do you think its an overengineering?

Reply:
The idea was indeed to avoid having one mega filler with 4000 lines of code, and to be able to work on remaining pieces more easily in multiple PRs. But maybe at the end this should be all done inside one, we could refacto this and merge them all into one We can discuss this tomorrow I guess